### ISSUE TITLE
Support emmet-ls

### ISSUE THREAD

**Issue by sei40kr** [emacs-lsp/lsp-mode#3381]
I want emmet-ls support in lsp-mode.

Actually I implemented on my own & it worked.

But `make docs` didn't work and couldn't update docs.
https://github.com/sei40kr/lsp-mode/commit/2bc0b1be9066c23a9ef00ea850fec9c65c1087c4

<details>
<summary>The results of $ make docs</summary>

```
$ make docs
make -C docs/ generate
make[1]: Entering directory '/home/sei40kr/ghq/github.com/sei40kr/lsp-mode/docs'
cask install

Error: void-variable (package-recipe--working-tree)
  debug-early-backtrace()
  debug-early(error (void-variable package-recipe--working-tree))
  byte-code(    \"
\"!#
        \"$     \"      \"\"#!\"!\"\"##$" [package-recipe--working-tree rcp package-build-working-dir package-recipe--upstream-url defmethod (rcp package-recipe) file-name-as-directory expand-file-name eieio-oref name (rcp package-recipe) url format url-format repo defalias package-git-recipe-p eieio-make-class-predicate package-git-recipe package-git-recipe--eieio-childp eieio-make-child-predicate package-git-recipe-child-p make-obsolete "use (cl-typep ... 'package-git-recipe) instead" "25.1" define-symbol-prop cl-deftype-satisfies eieio-defclass-internal (package-recipe) ((tag-regexp :initform "\\([0-9]\\{4\\}-[0-9]\\{2\\}-[0-9]\\{2\\} [0-9]\\{2\\}:[0-9]\\{2\\}:[0-9]\\{2\\}\\( [+-][0-9]\\{4\\}\\)?\\)")) nil] 8)
  require(package-recipe)
  load-with-code-conversion("/home/sei40kr/.config/emacs/.cask/29.0/bootstrap/package-build-2.4/package-build.el" "/home/sei40kr/.config/emacs/.cask/29.0/bootstrap/package-build-2.4/package-build.el" nil t)
  require(package-build)
  (if (featurep pkg) nil (if (package-installed-p pkg) nil (if package-archive-contents nil (package-refresh-contents)) (package-install pkg)) (require pkg))
  (let ((pkg (car --dolist-tail--))) (if (featurep pkg) nil (if (package-installed-p pkg) nil (if package-archive-contents nil (package-refresh-contents)) (package-install pkg)) (require pkg)) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (while --dolist-tail-- (let ((pkg (car --dolist-tail--))) (if (featurep pkg) nil (if (package-installed-p pkg) nil (if package-archive-contents nil (package-refresh-contents)) (package-install pkg)) (require pkg)) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (let ((--dolist-tail-- deps)) (while --dolist-tail-- (let ((pkg (car --dolist-tail--))) (if (featurep pkg) nil (if (package-installed-p pkg) nil (if package-archive-contents nil (package-refresh-contents)) (package-install pkg)) (require pkg)) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  (let (package-alist package-archive-contents (load-path load-path) (package-archives '(("gnu" . "https://elpa.gnu.org/packages/") ("melpa" . "https://stable.melpa.org/packages/"))) (package-user-dir cask-bootstrap-dir) (deps '(s f commander git epl shut-up cl-lib cl-generic package-build eieio ansi))) (package-initialize) (if (version= emacs-version "24.4") (progn (require 'tls) (custom-set-default 'tls-program '("openssl s_client -connect %h:%p -no_ssl3 -no_ssl2 -ign_eof")) (advice--add-function :override (cons #'(lambda nil (symbol-function 'gnutls-available-p)) #'(lambda (gv--val) (fset 'gnutls-available-p gv--val))) #'ignore nil))) (if (version< emacs-version "25.1") (progn (if (package-installed-p 'cl-lib) nil (if package-archive-contents nil (package-refresh-contents)) (package-install 'cl-lib)) (require 'package-recipe (expand-file-name "package-recipe-legacy" cask-directory)) (require 'package-build (expand-file-name "package-build-legacy" cask-directory)))) (let ((--dolist-tail-- deps)) (while --dolist-tail-- (let ((pkg (car --dolist-tail--))) (if (featurep pkg) nil (if (package-installed-p pkg) nil (if package-archive-contents nil (package-refresh-contents)) (package-install pkg)) (require pkg)) (setq --dolist-tail-- (cdr --dolist-tail--))))))
  load-with-code-conversion("/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-bootstrap.el" "/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-bootstrap.el" nil t)
  require(cask-bootstrap "/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-bootstrap")
  #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_102>(#<buffer  *load*> "/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-cli.el")
  load-with-code-conversion("/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-cli.el" "/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-cli.el" nil t #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_102>)
  command-line--load-script("/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-cli.el")
  command-line-1(("-scriptload" "/nix/store/ssv6avi9d8m9xa8dqp6nz8qbalv2f12b-cask-0.8.7/share/emacs/site-lisp/cask/cask-cli.el" "--" "install"))
  command-line()
  normal-top-level()
Symbol’s value as variable is void: package-recipe--working-tree
make[1]: *** [Makefile:4: deps] Error 255
make[1]: Leaving directory '/home/sei40kr/ghq/github.com/sei40kr/lsp-mode/docs'
make: *** [Makefile:110: docs] Error 2
```

</details>

<details>
<summary>The result of M-x (emacs-version)</summary>

```
GNU Emacs 29.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.31, cairo version 1.16.0)
```

</details>

**Comment by yyoncho** [comment]
I doubt that this commit will cause issues. Feel free to turn it into a PR. @ericdallo is more familiar with docs generation.

**Comment by ericdallo** [comment]
Never saw that exception when generating docs, maybe something with latest emacs? I'll try it locally to double check

**Comment by sei40kr** [comment]
Implemented as #3384.
